gap/Proset.gi

```diff
@@ -7,32 +7,36 @@
 SetInfoLevel( InfoLocales, 1 );
 
 ##
-InstallMethod( UniqueMorphism,
-        "for two objects in a thin category",
-        [ IsObjectInThinCategory, IsObjectInThinCategory ],
+InstallMethodForCompilerForCAP( UniqueMorphism,
+        "for a proset category and two objects in a thin category",
+        [ IsProsetOrPosetOfCapCategory, IsObjectInThinCategory, IsObjectInThinCategory ],
 
-  function( A, B )
-    local C, u;
+  function( C, A, B )
     
-    C := CapCategory( A );
+    #% CAP_JIT_DROP_NEXT_STATEMENT
```

Comment:
`if`s with a single `Error` statement are automatically dropped by `CompilerForCAP`, so `CAP_JIT_DROP_NEXT_STATEMENT` should not be necessary here.